src/index.js

```diff
@@ -0,0 +1,127 @@
+  }
+  if (typeof parsed !== 'object') { throw new Error(stderr || stdout); }
+  return parsed;
+};
+
+const forwardRubocopToLinter =
+  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
+    const [message, url] = rawMessage.split(/ \((.*)\)/, 2);
+    const { line, column, length } = location || { line: 1, column: 1, length: 0 };
+    return {
+      type: ['refactor', 'convention', 'warning'].includes(severity) ? 'Warning' : 'Error',
+      html: formatMessage({ cop_name, message, url }),
+      filePath,
+      range: [[line - 1, column - 1], [line - 1, (column + length) - 1]],
+    };
+  };
+
+export default {
+  activate() {
+    this.subscriptions = new CompositeDisposable();
+
+    // Register fix command
+    this.subscriptions.add(
+      atom.commands.add('atom-text-editor', {
+        'linter-rubocop:fix-file': async () => {
+          const textEditor = atom.workspace.getActiveTextEditor();
+          const filePath = textEditor.getPath();
+          const command = this.command
+                              .split(/\s+/)
+                              .filter(i => i)
+                              .concat(DEFAULT_ARGS, '--auto-correct', filePath);
+
+          if (!atom.workspace.isTextEditor(textEditor) || textEditor.isModified()) {
+            // Abort for invalid or unsaved text editors
+            return atom.notifications.addError('Linter-Rubocop: Please save before fixing');
+          }
+
+          const cwd = path.dirname(helpers.find(filePath, '.'));
+          const { stdout, stderr } = await helpers.exec(command[0], command.slice(1), { cwd, stream: 'both' });
+          const { summary: { offense_count: offenseCount } } = parseFromStd(stdout, stderr);
+          return offenseCount === 0 ?
+            atom.notifications.addInfo('Linter-Rubocop: No fixes were made') :
+            atom.notifications.addSuccess(`Linter-Rubocop: Fixed ${pluralize('offenses', offenseCount, true)}`);
+        },
+      }),
+    );
+
+    // Config observers
+    this.subscriptions.add(
+      atom.config.observe('linter-rubocop.command', (value) => {
+        this.command = value;
+      }),
+    );
+    this.subscriptions.add(
+      atom.config.observe('linter-rubocop.disableWhenNoConfigFile', (value) => {
+        this.disableWhenNoConfigFile = value;
+      }),
+    );
+  },
+
+  deactivate() {
+    this.subscriptions.dispose();
+  },
+
+  provideLinter() {
+    return {
+      name: 'RuboCop',
+      grammarScopes: [
+        'source.ruby',
+        'source.ruby.rails',
+        'source.ruby.rspec',
+        'source.ruby.chef',
+      ],
+      scope: 'file',
+      lintOnFly: true,
+      lint: async (editor) => {
+        const filePath = editor.getPath();
+        const command = this.command
+                            .split(/\s+/)
+                            .filter(i => i)
+                            .concat(DEFAULT_ARGS, '--stdin', filePath);
+        if (this.disableWhenNoConfigFile === true) {
```

Comment:
Might as well move this above the `command` parsing, since if this check fails there is no need to do that work (little as it may be).